archunit-tooling-test/build.gradle

```diff
@@ -0,0 +1,191 @@
+            runtimeClasspath += main.runtimeClasspath
+        }
+    }
+    main {
+        java {
+            compileClasspath += project.sourceSets.testFixtures.output
+            runtimeClasspath += project.sourceSets.testFixtures.output
+        }
+    }
+    test {
+        java {
+            compileClasspath += project.sourceSets.testFixtures.output
+            runtimeClasspath += project.sourceSets.testFixtures.output
+        }
+    }
+}
+
+copy {
+    into new File(project.sourceSets.testFixtures.output.resourcesDir, '@sources')
+    from(project.sourceSets.testFixtures.java) {
+        include '**/*.java'
+    }
+}
+
+processTestFixturesResources {
+    filesMatching(['**/pom.xml', '**/build.gradle']) {
+        filter ReplaceTokens, tokens: [
+                "junitDependency.version": project.property("archunit.version")
+        ]
+    }
+}
+
+dependencies {
+    implementation dependency.junit5JupiterEngine
+    implementation dependency.junit5VintageEngine
+    implementation dependency.junitPlatform
+    implementation dependency.guava
+    implementation 'org.junit.platform:junit-platform-launcher:1.8.2'
+    implementation project(path: ':archunit-junit5-engine')
+    implementation 'com.github.stefanbirkner:system-lambda:1.2.1'
+    implementation 'org.apache.maven.shared:maven-invoker:3.1.0'
+    implementation 'org.apache.maven.surefire:surefire-report-parser:3.0.0-M5'
+    implementation 'commons-io:commons-io:2.11.0'
+    implementation 'org.gradle:gradle-tooling-api:7.3-20210825160000+0000'
+
+    compileOnly 'com.github.spotbugs:spotbugs-annotations:4.5.3'
+
+    testFixturesImplementation dependency.junitPlatform
+    testFixturesImplementation dependency.junit5JupiterEngine
+    testFixturesImplementation dependency.junit5JupiterApi
+    testFixturesImplementation project(path: ':archunit')
+    testFixturesImplementation project(path: ':archunit-junit5-engine-api')
+    testFixturesImplementation project(path: ':archunit-junit5-api')
+    testFixturesImplementation project(path: ':archunit-junit5-engine')
+    testFixturesImplementation project(path: ':archunit-junit4')
+
+    testImplementation dependency.assertj
+    testImplementation 'org.junit.jupiter:junit-jupiter-params:5.8.2'
+    testImplementation 'org.junit-pioneer:junit-pioneer:1.6.1'
+}
+
+String getMavenRepoPath(Project project) {
+    def pubs = project.publishing.publications.findAll() as Set<MavenPublication>
+    def pub = pubs[0]
+    if (!pub){
+        return null
+    }
+    String groupPath = pub.groupId.replace('.', '/')
+    return "$groupPath/$pub.artifactId"
+}
+
+def resolveAgainstMavenLocal(String path) {
+    new File(new File(repositories.mavenLocal().url), path)
+}
+
+def deleteSafely(File file) {
+    if (file.exists()) {
+        println "Deleting ${file.toURI().toString()}"
+        delete file
+    } else {
+        println "Could not find ${file.toURI().toString()}"
+    }
+}
+
+task removeFromMavenLocal {
```

Comment:
This task cleans the local Maven repo to make sure we are running the Surefire and Gradle tests with an up-to-date version of ArchUnit jars